openedx/core/djangoapps/credit/api/eligibility.py

```diff
@@ -225,6 +228,15 @@ def set_credit_requirement_status(username, course_key, req_namespace, req_name,
             )
 
     """
+    username = user.username
+    # Check whether user has credit enrollment. Only credit enrollments should be eligible for credit.
+    enrollment_mode, is_active = CourseEnrollment.enrollment_mode_for_user(user, course_key)
+    has_credit_enrollment = (CourseMode.is_credit_slug(enrollment_mode) and is_active)
```

Comment:
@schenedx @clintonb  Thank you for the suggestion. 
so the correct behaviour should be to check if a user has enrollment in the [verified modes](https://github.com/edx/edx-platform/blob/260f6b3a327834b3342b3c1e0b7b64ce16adef32/common/djangoapps/course_modes/models.py#L123-123)

Correct me if I am wrong.